middlewares.ts

```diff
@@ -23,47 +23,63 @@ export const authorizePls = async (
     res: Response,
     next: NextFunction
 ) => {
-    const authorizationHeader = req.headers.authorization;
-    let token;
+    const authorizationHeader: string | undefined = req.headers.authorization;
+    let token = "";
+    let response: AxiosResponse | undefined = undefined;
+    let url = "";
+
     if (authorizationHeader) {
         token = authorizationHeader.split(" ")[1];
     } else if (req.query.token) {
-        token = req.query.token;
+        token = req.query.token as string;
     }
 
-    if (!token || token.length === 0) {
+    if (token.length === 0) {
         res.sendStatus(401);
         return;
     }
 
     try {
-        const response = await axios.get(
-            `${configuration.LOGIN_API_URL}/verify/${token}.json?api_key=${configuration.LOGIN_API_KEY}`
-        );
+        url = `${configuration.LOGIN_API_URL}/verify/${token}.json?api_key=${configuration.LOGIN_API_KEY}`;
+        response = await axios.get(url);
+        if (response === undefined) {
+            res.status(500).send("Undefined response");
+            return;
+        }
         if (response.status !== 200) {
-            res.sendStatus(401);
+            res.status(response.status).send(response.data);
+            return;
+        }
+    } catch (err) {
+        response = (err as AxiosError).response;
+        if (response === undefined) {
+            res.status(500).send("Undefined response");
             return;
         }
```

Comment:
"undefined response" is a strange message to return
```suggestion
```

Reply:
Yeah, at that time I didn't really know what to send in that case.
My idea, however, was that if any of the APIs didn't send a response, the Pico server would send a response reporting that error. So I'm not sure if it's a good idea to delete this completely.

I changed the code so that it reflects more accurately what I wanted to convey, following some practices and formatting in the Axios website. The messages should be more descriptive now. I also changed the error code, for I believe that it should instead be 502 instead. The following description seems to match more accurately what I want to communicate to the frontend:
> The 502 Bad Gateway error is an [HTTP status code](https://www.lifewire.com/http-status-codes-2625907) that means that one server on the internet received an invalid response from another server.

What do you think about this change?